### DISCUSSION TITLE
classes/class_collisionshape2d

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#182] [User-contributed notes]
# classes/class_collisionshape2d

Inherits: Node2D< CanvasItem< Node< Object A node that provides a Shape2D to a CollisionObject2D parent. Description: A node that provides a Shape2D to a CollisionObject2D parent and allows to edit...

https://docs.godotengine.org/en/stable/classes/class_collisionshape2d.html

**Comment by kudorgyozo** [comment]
### Warning: Collision shape instances are cached between instances of the same scene.

If you don't want that there's the option ** Shape > Resource > Local to the scene. **

**Reply by nonchip** [reply]
there's no need to scream about this here, it's a feature of literally every resource in every scene, nothing to do with collisionshape nodes.

**Comment by defnotryan** [comment]
I am new to Godot and your comment solved a mystery for me! I guess I remember reading somewhere else at some point something about this caching, so perhaps I should have thought of it as a cause for setting the size on one collision shape's rectangle affected another collision shape's rectangle. But it would have taken me much longer to figure out had I not seen your comment. Thanks!

**Reply by defnotryan** [reply]
To clarify, in my mental model since my two scene instances had two different CollisionShape2D instances, each of those had their own Rectangle2D instance. However, those two CollisionShape2D instances shared once instance of Rectangle2D. So setting the size of that rectangle on one scene instance gave the then-surprising effect of changing the size of the Rectangle2D in the other scene instance. However, now that I realize they are the same Rectangle2D instance, everything makes sense. (And setting the "Local to scene" flag fixes it.)

I see how this behavior is desirable for many (most?) resources: often they will take a lot of memory and won't vary between scene instances. Textures, for example, benefit from this approach.

**Comment by bibizu** [comment]
Is the only way to change the one way collision direction by rotating the entire shape? I cannot find anything about it. It would make sense for it to be its own property to allow for easy editing.

Is there a reason for this? Right now, for example, you cannot make a CollisionShape2D with a RectangleShape2D be one way from a 45 degree angle (for whatever niche reason). It seems to work with a CollisionPolygon2D with the right rotation, however. But it makes working with one way collisions very tedious, as you cannot use the most obvious workflow for such a collision. I personally am having issues with segment shapes and one way collisions, which seems more applicable.

I will search some more, but I might make this into a proposal if no one else has.
